linalg/densemat.cpp

```diff
@@ -1329,6 +1329,20 @@ void DenseMatrix::CalcEigenvalues(real_t *lambda, real_t *vec) const
    }
 }
 
+real_t DenseMatrix::CalcConditionNumber() const
+{
+   if (width == 1) { return 1_r; }
```

Comment:
Doesn't this assume the matrix contains non-zero entries?

Reply:
One option is to add a check if the entry is zero and return `nan` in that case. Another option is to treat the singular matrix case as "undefined behavior" and just document that. Which option do you prefer -- either one works for me.

In the multi-D case, we can get `+/-inf` or `nan` right now.